### ISSUE TITLE
Issue with PR #7577 (WiFiGeneric: remove deprecated tcpip_adapter calls)

### ISSUE THREAD

**Issue by pablozg** [espressif/arduino-esp32#7963]
### Board

Heltec WiFi Kit 32

### Device Description

ESP32

### Hardware Configuration

No connection

### Version

latest master (checkout manually)

### IDE Name

PlatformIO

### Operating System

Windows 11

### Flash frequency

80Mhz

### PSRAM enabled

no

### Upload speed

115200

### Description

I have reports of some users having problems using a basic captive portal, where the connection to the AP cannot be reached.

Until this PR, everything was working well, no issues with the captive portal, personally I have not had any connection problems with it, but some users of my project reports issues.

After some trial and error, reversing this PR fixed the problem with the connection, can you give me any hints? 

It's a really hardware dependent bug and I only can do a few modifications and ask them to test to verify that they are resolved.

### Sketch

```cpp

This is the involved code:

        // err = esp_netif_dhcps_option(
        //     esp_netif,
        //     ESP_NETIF_OP_SET,
        //     ESP_NETIF_SUBNET_MASK,
        err = tcpip_adapter_dhcps_option(
            (tcpip_adapter_dhcp_option_mode_t)TCPIP_ADAPTER_OP_SET,
            (tcpip_adapter_dhcp_option_id_t)ESP_NETIF_SUBNET_MASK,
            (void*)&info.netmask.addr, sizeof(info.netmask.addr)
        );
		if(err){
        	log_e("DHCPS Set Netmask Failed! 0x%04x", err);
        	return err;
        }
        // err = esp_netif_dhcps_option(
        //     esp_netif,
        //     ESP_NETIF_OP_SET,
        //     ESP_NETIF_REQUESTED_IP_ADDRESS,
        err = tcpip_adapter_dhcps_option(
            (tcpip_adapter_dhcp_option_mode_t)TCPIP_ADAPTER_OP_SET,
            (tcpip_adapter_dhcp_option_id_t)REQUESTED_IP_ADDRESS,
            (void*)&lease, sizeof(dhcps_lease_t)
        );

With this is working again
```

### Debug Message

```plain
[265047][V][captivePortal.cpp:182] http_404_error_handler(): [Captive Portal] Request received to uri: /chat and type 3
[265047][I][captivePortal.cpp:189] http_404_error_handler(): [Captive Portal] Redirecting to http://192.168.4.1
[265058][I][captivePortal.cpp:194] http_404_error_handler(): [Captive Portal] Redirecting to root from 404 error handler
[315961][E][captivePortal.cpp:78] savedata_post_handler(): [Captive Portal] savedata
[315961][V][captivePortal.cpp:110] savedata_post_handler(): [Captive Portal] Data: wifis=%7B%22SSID%22%3A%22FAN%20WIFI%22%2C%22BSSID%22%3A%22E4%3A0E%3AEE%3AB0%3A86%3AB9%22%7D&password=XXXXXXXXXXXXXXXXXXX&host=XXXXXXX
[315976][V][captivePortal.cpp:113] savedata_post_handler(): [Captive Portal] Found URL query parameter => wifis={"SSID":"FAN WIFI","BSSID":"E4:0E:EE:B0:86:B9"}
[315990][I][webLanHandlers.cpp:25] parse_ssid(): [HTTP Index Handlers] SSID Json: {"SSID":"FAN WIFI","BSSID":"E4:0E:EE:B0:86:B9"}
[316002][V][captivePortal.cpp:119] savedata_post_handler(): [Captive Portal] Found URL query parameter => password=123456789101112
[316013][V][captivePortal.cpp:124] savedata_post_handler(): [Captive Portal] Found URL query parameter => host=XXXXX
[316026][V][WiFiGeneric.cpp:97] set_esp_interface_ip(): Configuring Station static IP: 0.0.0.0, MASK: 0.0.0.0, GW: 0.0.0.0
[316043][I][captivePortal.cpp:131] savedata_post_handler(): [Captive Portal] Trying to connect to Wi-Fi with ssid: FAN WIFI and pass: XXXXXXXXXXXXXXXXX
[317047][I][captivePortal.cpp:138] savedata_post_handler(): [Captive Portal] Trying to connect to Wi-Fi...
[318047][I][captivePortal.cpp:138] savedata_post_handler(): [Captive Portal] Trying to connect to Wi-Fi...
[319044][V][WiFiGeneric.cpp:362] _arduino_event_cb(): STA Disconnected: SSID: FAN WIFI, BSSID: e4:0e:ee:b0:86:b9, Reason: 15
[319044][D][WiFiGeneric.cpp:931] _eventCallback(): Arduino Event: 5 - STA_DISCONNECTED
[319047][I][captivePortal.cpp:138] savedata_post_handler(): [Captive Portal] Trying to connect to Wi-Fi...
[319052][W][WiFiGeneric.cpp:955] _eventCallback(): Reason: 15 - 4WAY_HANDSHAKE_TIMEOUT
[319069][D][WiFiGeneric.cpp:975] _eventCallback(): WiFi Reconnect Running
[319077][V][WiFiGeneric.cpp:97] set_esp_interface_ip(): Configuring Station static IP: 0.0.0.0, MASK: 0.0.0.0, GW: 0.0.0.0
[319095][D][wifiFunctions.cpp:270] WiFiEvent(): [WIFI] [WiFi-event] event: 5
```
```


### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by mrengineer7777** [comment]
Constants like that are going to depend on the version of LWIP and IDF.  It is critical that arduino-esp32 and IDF use compatible versions.  More info needed.

**Comment by pablozg** [comment]
Hi, I'm using the master branch of ArduinoESP32

This is my platformio section:
```
platform                    = 
                              https://github.com/platformio/platform-espressif32.git

platform_packages           = 
                              framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git
                              espressif/toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
```

This is the console output on build:

```
PLATFORM: Espressif 32 (6.0.1+sha.6bfea50) > Heltec WiFi Kit 32
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 2.0.6+sha.0d84018
 - tool-esptoolpy @ 1.40500.0 (4.5.0)
 - tool-openocd-esp32 @ 2.1100.20220706 (11.0)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
```

And the Esp32 info on running firmware:
```
ESP model ESP32-D0WDQ6, Revision 1, SDK version v4.4.3, Core: 160Mhz, Xtal: 26Mhz, APB: 80Mhz, MAC Address: 24:6F:28:2B:02:00, Heap Size: 289732, Free Heap: 242692
``` 

To try any solution or make test I must compile and send the binary to the user with issues and write here the feedback.

Thanks for your help.

**Comment by mrengineer7777** [comment]
Are you using "Arduino as IDF component"?  If not, then add `framework = arduino`.  The Arduino version should be 2.0.7.  Normally you shouldn't have to specify the framework and packages under platform_packages.  Try commenting that out.

Consider trying the custom branch from TASMOTA:
```
platform = https://github.com/tasmota/platform-espressif32/releases/download/2023.02.00/platform-espressif32.zip
framework = arduino
```

**Comment by pablozg** [comment]
> Are you using "Arduino as IDF component"? If not, then add `framework = arduino`. The Arduino version should be 2.0.7. Normally you shouldn't have to specify the framework and packages under platform_packages. Try commenting that out.
> 
> Consider trying the custom branch from TASMOTA:
> 
> ```
> platform = https://github.com/tasmota/platform-espressif32/releases/download/2023.02.00/platform-espressif32.zip
> framework = arduino
> ```

Hi, yes I'm using arduino as framework, I don't know why at compile time is showing 2.0.6 version but the revision match the last commit from master branch (0d84018).

I will give a try to the tasmota branch, but my project is an intesive use  tcp/ip app, and I use the master branch because I get more stability than with the 2.0.6 version.

**Comment by pablozg** [comment]
@mrengineer7777 Hi again, I try it, but I can use it, by the headers limit , I get "Header fields are too long for server to interpret"

**Comment by mrengineer7777** [comment]
Bummer! I know the latest TASMOTA includes support for IPv6. Perhaps it
reduces the header length.

**Comment by Jason2866** [comment]
@pablozg  Can you try this build https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/1243/framework-arduinoespressif32-lwip_timeout-ed6742e7f0.zip
The Platformio the setup is
```
platform = https://github.com/tasmota/platform-espressif32/releases/download/2023.02.00/platform-espressif32.zip
platform_packages = framework-arduinoespressif32 @ https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/1243/framework-arduinoespressif32-lwip_timeout-ed6742e7f0.zip
```
Reverted lwip in IDF to an older version in this build, since we have timeout issues with DHCP in special cases. Wifi sometimes does not get an IP address when an Ethernet adapter is configured and active at the same time. With this build (not yet used for Tasmota!) the issue is solved. Actual IDF lwip shortened timeout time for DHCP request. To short sometimes!

**Comment by Jason2866** [comment]
> Bummer! I know the latest TASMOTA includes support for IPv6. Perhaps it reduces the header length.
> […](#)

We are checking, should not be the case

**Comment by pablozg** [comment]
> 

I have tried it, but I still getting the "Header fields are too long for server to interpret" message when try to open the webpage.

**Comment by Jason2866** [comment]
Dont know where "Header fields are too long for server to interpret" is coming from.

**Comment by pablozg** [comment]
> Dont know where "Header fields are too long for server to interpret" is coming from.
> There is nothing changed in "header length"

In the master branch the idf sdk was compiled with CONFIG_HTTPD_MAX_REQ_HDR_LEN set to 1024 (\tools\sdk\esp32\sdkconfig), I checked in your framework and is set to 512.

**Comment by Jason2866** [comment]
@pablozg YOu can try this build https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/1254/framework-arduinoespressif32-lwip_timeout-ed6742e7f0.zip
-> CONFIG_HTTPD_MAX_REQ_HDR_LEN set to 1024

**Comment by pablozg** [comment]
@Jason2866 Hi, now the web page is working ok, I will send the binary to the user with issues and write back his feedback.

**Comment by pablozg** [comment]
@Jason2866 Have you disabled the use of the "ESP_LOG" functions? All messages using ESP_LOG functions aren't printed on serial port., only are printed with "log_" functions like "log_i".

**Comment by Jason2866** [comment]
@pablozg Yes, everything which needs flash space is removed. There is a reason why we use our own fork. Trimmed for our need.

**Comment by pablozg** [comment]
Anyway, while I wait for his response, what it's the difference between ```esp_netif_dhcps_option``` and ```tcpip_adapter_dhcps_option``` to cause this issue?

**Comment by mrengineer7777** [comment]
This repository also disables ESP_LOG functions in favor of log_i(), log_e(), etc.

**Comment by pablozg** [comment]
Hi, I have feedback from the user with issues.

1- Tested arduino 2.0.7 release, the esp32 can't connect to the ap with long response time.

2- Tested arduino 2.0.7 release with reverted PR, can connect without issues.

3- Tested 2.0.7 Tasmota version, can connect without issue.

I asked him to still using the tasmota version in order to test the stability, I will report here as soon as I get his feedback.

I don't know why, but I only can get working again if I revert the PR.

In my case I have tested it with 3 different routers, I can confirm that with or without the PR can connect correctly in all cases. 

I have tested master branch, 2.0.7 release and tasmota version.
